tests/unit/test_exec.py

```diff
@@ -0,0 +1,54 @@
+__copyright__ = 'Copyright (c) 2020-2021 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+from pathlib import Path
+
+import pytest
+from jina import Document, DocumentArray, Executor
+
+from video_loader import VideoLoader
+
+
+@pytest.fixture(scope="module")
+def encoder() -> VideoLoader:
+    return VideoLoader()
+
+
+def test_config():
+    ex = Executor.load_config(str(Path(__file__).parents[2] / 'config.yml'))
+    assert ex.fps == 1
+    assert ex.max_num_frames == 50
+
+
+def test_no_docucments(encoder: VideoLoader):
+    docs = DocumentArray()
+    encoder.extract(docs=docs)
+    assert len(docs) == 0  # SUCCESS
+
+
+def test_docs_no_uris(encoder: VideoLoader):
+    docs = DocumentArray([Document()])
+
+    with pytest.raises(ValueError, match='No uri'):
```

Comment:
Should it really raise when no uri? It should just log a warning

Reply:
I think it would be best if you just `logger.warning` or `logger.error` instead, with the Document id or something. A stack trace would be a bit too much in the output.